sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SwaggerMethodParser.java

```diff
@@ -322,38 +306,38 @@ public String setPath(Object[] methodArguments) {
 
     /**
      * Get the headers that have been added to this value based on the provided method arguments.
+     *
      * @param swaggerMethodArguments The arguments that will be used to create the headers' values.
      * @return An Iterable with the headers.
      */
     public Iterable<HttpHeader> setHeaders(Object[] swaggerMethodArguments) {
         final HttpHeaders result = new HttpHeaders(headers);
 
-        if (headerSubstitutions != null) {
```

Comment:
`headerSubstituations` is instantiated to an empty `List` and will only be added to, will never be null.